src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -174,6 +174,19 @@ private IAIState serveFoodToCitizen()
 
         if (InventoryUtils.isItemHandlerFull(handler))
         {
+            if (!citizenToServe.isEmpty())
+            {
+                final int foodSlot = InventoryUtils.findFirstSlotInItemHandlerWith(worker.getInventoryCitizen(), ItemStackUtils.CAN_EAT);
+                if (foodSlot != -1)
+                {
+                    final ItemStack stack = worker.getInventoryCitizen().extractItem(foodSlot, 1, false);
```

Comment:
Please add in a safety check that checks if serving the first citizen is actually usefull.

Reply:
What do you mean?

Reply:
I mean, please check if it is useful to feed the first citizen
Or if it is more useful to feed another citizen with that piece of meal.